pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -321,17 +375,38 @@ func (p *provider) Create(machine *clusterv1alpha1.Machine, _ *cloudprovidertype
 	// because its pod got deleted
 	// The secret has an ownerRef on the VMI so garbace collection will take care of cleaning up
 	terminationGracePeriodSeconds := int64(30)
-	userDataSecretName := fmt.Sprintf("userdata-%s-%s", machine.Name, strconv.Itoa(int(time.Now().Unix())))
-	requestsAndLimits, err := parseResources(c.CPUs, c.Memory)
+	userDataSecretName := fmt.Sprintf("userdata-%s-%s", c.VirtualMachineName, strconv.Itoa(int(time.Now().Unix())))
+
+	resourceRequirements := kubevirtv1.ResourceRequirements{}
+	labels := map[string]string{"kubevirt.io/vm": c.VirtualMachineName}
+
+	sigClient, err := client.New(c.RestConfig, client.Options{})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get kubevirt client: %v", err)
 	}
+	ctx := context.Background()
 
-	var (
-		pvcRequest     = corev1.ResourceList{corev1.ResourceStorage: c.PVCSize}
-		dataVolumeName = machine.Name
+	// Add VMIPreset label if specified
+	if c.FlavorName != "" {
+		vmiPreset := kubevirtv1.VirtualMachineInstancePreset{}
+		if err := sigClient.Get(ctx, types.NamespacedName{Namespace: c.Namespace, Name: c.FlavorName}, &vmiPreset); err != nil {
```

Comment:
Until we have implemented https://app.zenhub.com/workspaces/work-in-progress---team-caffeine-overflow-6166fa570192a800141b6db9/issues/kubermatic/kubermatic/8490
We have a fetch the `Presets` from `default` namespace